components/ledger/ledger-utxo-token-cache/src/main/kotlin/net/corda/ledger/utxo/token/cache/entities/internal/TokenPoolCacheImpl.kt

```diff
@@ -1,14 +1,29 @@
 package net.corda.ledger.utxo.token.cache.entities.internal
 
+import com.github.benmanes.caffeine.cache.Caffeine
 import net.corda.ledger.utxo.token.cache.entities.TokenCache
 import net.corda.ledger.utxo.token.cache.entities.TokenPoolCache
 import net.corda.ledger.utxo.token.cache.entities.TokenPoolKey
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
 
-class TokenPoolCacheImpl : TokenPoolCache {
+class TokenPoolCacheImpl(expiryPeriod: Duration) : TokenPoolCache {
 
-    private val cache = mutableMapOf<TokenPoolKey, TokenCache>()
+    private val cache = if (expiryPeriod == Duration.ZERO) {
+        Caffeine.newBuilder().build<TokenPoolKey, TokenCache>()
+    } else {
+        Caffeine.newBuilder()
+            .expireAfterWrite(expiryPeriod.toLong(DurationUnit.MILLISECONDS), TimeUnit.MILLISECONDS)
```

Comment:
We can pass a Duration object into this method. We just need to use the Java Duration, not the Kotlin one.

```suggestion
            .expireAfterWrite(expiryPeriod)
```

Reply:
Tried it locally and it was a bit involved. I've raised this PR against your branch:

https://github.com/corda/corda-runtime-os/pull/5909

Reply:
Thanks. I was going through those changes now.